controllers/releaseplan/enqueue_status.go

```diff
@@ -0,0 +1,86 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package releaseplan
+
+import (
+	"context"
+	"strings"
+
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	crtHandler "sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var log = logf.Log.WithName("event_handler")
+
+// EnqueueForReleasePlanAdmissionStatus enqueues Request containing the Name and Namespace specified in the
+// Status.ReleasePlans[] of the ReleasePlanAdmission that is the source of the Event. The source of the event
+// triggers reconciliation of the parent resource.
+type EnqueueForReleasePlanAdmissionStatus struct {
+	Type schema.GroupKind
+}
+
+var _ crtHandler.EventHandler = &EnqueueForReleasePlanAdmissionStatus{}
+
+// Create implements EventHandler
+func (e *EnqueueForReleasePlanAdmissionStatus) Create(_ context.Context, evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	// A freshly created ReleasePlanAdmission won't have any ReleasePlans in its status
+}
+
+// Update implements EventHandler
+func (e *EnqueueForReleasePlanAdmissionStatus) Update(_ context.Context, evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	e.queueReleasePlansInStatus(evt.ObjectOld, q)
+	e.queueReleasePlansInStatus(evt.ObjectNew, q)
+}
+
+// Delete implements EventHandler
+func (e *EnqueueForReleasePlanAdmissionStatus) Delete(_ context.Context, evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	e.queueReleasePlansInStatus(evt.Object, q)
+}
+
+// Generic implements EventHandler
+func (e *EnqueueForReleasePlanAdmissionStatus) Generic(_ context.Context, evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	e.queueReleasePlansInStatus(evt.Object, q)
+}
+
+// queueReleasePlansInStatus enqueues a reconcile request for each ReleasePlan in a ReleasePlanAdmission's status.
+func (e *EnqueueForReleasePlanAdmissionStatus) queueReleasePlansInStatus(object metav1.Object, q workqueue.RateLimitingInterface) {
+	if releasePlanAdmission, ok := object.(*v1alpha1.ReleasePlanAdmission); ok {
+		for _, releasePlan := range releasePlanAdmission.Status.ReleasePlans {
+			q.Add(reconcile.Request{NamespacedName: parseNamespacedName(releasePlan.Name)})
+		}
+	}
+}
+
+// parseNamespacedName parses the provided string to extract the namespace and name into a
+// types.NamespacedName.
+func parseNamespacedName(namespacedNameString string) types.NamespacedName {
+	values := strings.SplitN(namespacedNameString, "/", 2)
+
+	switch len(values) {
+	case 1:
+		return types.NamespacedName{Name: values[0]}
```

Comment:
Why to support this? The CRD only supports valid namespaced names. If the len is not 2, it should be ignored.

Reply:
fixed